### PR TITLE
Fix FastAPI issues

### PR DESCRIPTION
**Pull request by antonpirker**
This issue is a collection of bug fixes for the few bugs that where discovered from the FastAPI alpha release.

* Patch FastAPI `get_request_handler` to extract request body data (fixes the infinite loop when doing form posts in FastAPI) #1514
* Fix patching exception handlers in `_sentry_patched_exception_handler`(fixes the error on non existent routes) #1514
* Made Starlette middleware patching idempotent to prevent patching functions twice (no GitHub issue for that)
* Patch `request_response` in Starlette to extract request body data  (fixes the infinite loop when doing form posts in Starlette) #1514
* Starlette has async and sync "views" (or route handlers) in the sync ones it is not easily possible to get the request body data (because those functions are async) so we just get the cookies form the sync route handlers and not the body data. (no GitHub issue for that)

### CODE REVIEW

File: sentry_sdk/integrations/fastapi.py [hunk 1]
```diff
@@ -82,6 +83,60 @@ def _set_transaction_name_and_source(event, transaction_style, request):
     event["transaction_info"] = {"source": SOURCE_FOR_STYLE[transaction_style]}
 
 
+def patch_get_request_handler():
```
Comment (sl0thentr0py) [inline sentry_sdk/integrations/fastapi.py:86]: if we patch the request handler, do we need `SentryFastApiMiddleware` anymore?

Comment (sl0thentr0py) [reply sentry_sdk/integrations/fastapi.py:86]: same question for `SentryStarletteMiddleware`

**Review by sl0thentr0py — COMMENTED** [review]
question before detailed review

File: sentry_sdk/integrations/fastapi.py (hunk 1 above)
Comment (antonpirker) [reply sentry_sdk/integrations/fastapi.py:86]: Right now the `SentryFastApiMiddleware` is only setting the transaction name and source. But I guess we can move this into the request handler and get rid of the middleware!

Comment (sl0thentr0py) [reply sentry_sdk/integrations/fastapi.py:86]: yea I'd prefer only one place to set transaction name/source.

Comment (antonpirker) [reply sentry_sdk/integrations/fastapi.py:86]: There were some differences in Starlette and FastAPI. Can't remember exactly what, but something with routing or handling the routes. So I think we need to set the transaction name and source in both integrations to make sure we have the correct one in FastAPI projects. 
But we can get rid of the `SentryFastApiMiddleware`.

Comment (sl0thentr0py) [reply sentry_sdk/integrations/fastapi.py:86]: yes I meant one place per Starlette/Fastapi integration.
So I think we can get rid of both `SentryFastApiMiddleware` and `SentryStarletteMiddleware` in theory.

**Comment by MrSalman333** [comment]
when will this be released? 

thanks for the great work

**Comment by antonpirker** [comment]
We are working on the final touches. Should not take that long anymore.

File: sentry_sdk/integrations/fastapi.py (hunk 1 above)
Comment (antonpirker) [reply sentry_sdk/integrations/fastapi.py:60]: I have now cleaned up the code and the transaction name/source are only set once in fastapi and once in starlette integration.

**Review by sl0thentr0py — APPROVED** [review]

**Pull request merged**